action/copy.go

```diff
@@ -193,6 +192,18 @@ func CopyNotSeekableFileToRemote(r io.Reader, objectKey string) (total int64, er
 	return total, nil
 }
 
+// CopySeekableFileToRemote will copy a seekable file to remote.
+func CopySeekableFileToRemote(r *os.File, objectKey string) (total int64, err error) {
+	defer r.Close()
+	fileInfo, err := r.Stat()
+	if err != nil {
+		log.Errorf("Get file size failed [%v]", err)
+		return 0, constants.ErrorFileSizeInvalid
+	}
+	contexts.ExpectSize = fileInfo.Size()
+	return CopyNotSeekableFileToRemote(r, objectKey)
```

Comment:
`CopyNotSeekableFileToRemote` is optimized for stdin actually, and we don't need to cache content while we uploading a seekable file. Let's open file multiple times and use `SectionReader` to read data.